src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParser.cs

```diff
@@ -1079,20 +1066,16 @@ private void ScanBlank()
 
                 if ((_options & RegexOptions.IgnorePatternWhitespace) != 0 && _pos < _pattern.Length && _pattern[_pos] == '#')
                 {
-                    while (_pos < _pattern.Length && _pattern[_pos] != '\n')
-                    {
-                        _pos++;
-                    }
+                    _pos = _pattern.IndexOf('\n', _pos);
+                    if (_pos == -1)
+                        _pos = _pattern.Length;
```

Comment:
```C#
if (_pos < 0)
{
    _pos = _pattern.Length;
}
```

Reply:
Cheaper?

Reply:
A bit, and also matches our style preferences and the style of the rest of the code better.